src/helpers/Workspace.hpp

```diff
@@ -25,6 +34,7 @@ class CWorkspace {
     int             m_iPrevWorkspaceID = -1;
     bool            m_bHasFullscreenWindow = false;
     eFullscreenMode m_efFullscreenMode = FULLSCREEN_FULL;
+    eOrientation    m_orientation = ORIENTATION_LEFT;
```

Comment:
this has no place in `CWorkspace`, as it's a layout-specific thing.

Consider making a separate class / struct in MasterLayout for workspace-specific data, and storing it there.

e.g.
```cpp
struct stuff {
    int workspaceid = -1; 
    eorient orient = 0;
 }

...
    // in the layout, a private field
    std::vector<stuff> m_vStuff;
    stuff getStuffForWorkspace(int id);
```

Reply:
Right, I too felt it was a bit misplaced there, I simply took the easiest path. I'll try improve it as you suggested.

Reply:
I implemented this now, one thing I'm still a bit unsure of is the on-the-fly struct creation I do in https://github.com/hyprwm/Hyprland/blob/444be966bca1022fc77c63e59a96528b1e4a8f1e/src/layout/MasterLayout.cpp#L40